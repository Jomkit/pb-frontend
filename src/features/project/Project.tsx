
import { Link } from 'react-router-dom';
import { IProject } from '../../types';

const Project = ({project}: {project: IProject}) => {
  const {id, name, note} = project;

  return (
    <Link to={`/projects/${id}`}>
      <div className='w-full mx-auto p-2 mb-2 border-solid border-2 border-black rounded-md bg-white' id={id}>
          <div className="header">
              <h1 className="text-3xl text-bold">{name}</h1>
              <h3>Description: {note}</h3>
          </div>
      </div>
    </Link>
  )
}

export default Project 