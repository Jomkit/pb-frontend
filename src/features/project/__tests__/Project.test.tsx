import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import Project from "../Project"

it('should render without crashing', () => { 
    const testProject = {
        id: "1",
        name: "testProject",
        note: "testNote"
    }
    render(
        <MemoryRouter>
            <Project project={testProject} />
        </MemoryRouter>
    )
})
it("should match snapshot", () => {
    const testProject = {
        id: "1",
        name: "testProject",
        note: "testNote"
    }
    const { asFragment } = render(
        <MemoryRouter>
            <Project project={testProject} />
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})