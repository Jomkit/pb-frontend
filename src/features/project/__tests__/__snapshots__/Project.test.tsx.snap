// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <a
    href="/projects/1"
  >
    <div
      class="w-full mx-auto p-2 mb-2 border-solid border-2 border-black rounded-md bg-white"
      id="1"
    >
      <div
        class="header"
      >
        <h1
          class="text-3xl text-bold"
        >
          testProject
        </h1>
        <h3>
          Description: testNote
        </h3>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
